jupyterhub/spawners.py

```diff
@@ -87,51 +101,49 @@ def start(self, *args, **kwargs):
         self.log.debug("user options: {}".format(self.user_options))
 
         auth_state = yield self.user.get_auth_state()
-        assert "access_token" in auth_state
+        if GITLAB_AUTH:
+            assert "access_token" in auth_state
+            oauth_token = auth_state["access_token"]
+        else:
+            oauth_token = None
 
         options = self.user_options
         namespace = options.get("namespace")
         project = options.get("project")
         self.image = options.get("image")
+        self.cmd = "jupyterhub-singleuser"
 
         url = os.getenv("GITLAB_URL", "http://gitlab.renku.build")
 
+        gl = gitlab.Gitlab(url,
+            api_version=4,
+            oauth_token=oauth_token
+        )
+        gl_project = gl.projects.get("{0}/{1}".format(namespace, project))
+
         # check authorization against GitLab
-        gl = gitlab.Gitlab(url, api_version=4, oauth_token=auth_state["access_token"])
+        if GITLAB_AUTH:
+            try:
+                gl.auth()
+                self.gl_user = gl.user
 
-        try:
-            gl.auth()
-            gl_project = gl.projects.get("{0}/{1}".format(namespace, project))
-            self.gl_user = gl.user
+            except Exception as e:
+                self.log.error(e)
+                raise web.HTTPError(401, "Not logged in with GitLab.")
 
             # gather project permissions for the logged in user
-            permissions = gl_project.attributes["permissions"]
-            access_level = max(
-                [x[1].get("access_level", 0) for x in permissions.items() if x[1]]
-            )
-            self.log.debug(
-                "access level for user {username} in "
-                "{namespace}/{project} = {access_level}".format(
-                    username=self.user.name,
-                    namespace=namespace,
-                    project=project,
-                    access_level=access_level,
-                )
-            )
```

Comment:
should we keep logging similar info? It may be useful to have something like that when debugging

Reply:
Good point, added it back.